SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditor.cpp

```diff
@@ -50,87 +52,110 @@ bool FSpatialGDKEditor::GenerateSchema(bool bFullScan)
 		}
 	}
 
-	bSchemaGeneratorRunning = true;
-
-	// 80/10/10 load assets / gen schema / garbage collection.
-	FScopedSlowTask Progress(100.f, LOCTEXT("GeneratingSchema", "Generating Schema..."));
-	Progress.MakeDialog(true);
-
-	RemoveEditorAssetLoadedCallback();
-
 	if (Schema::IsAssetReadOnly(SpatialConstants::SCHEMA_DATABASE_FILE_PATH))
 	{
-		bSchemaGeneratorRunning = false;
 		return false;
 	}
 
-	if (!Schema::LoadGeneratorStateFromSchemaDatabase(SpatialConstants::SCHEMA_DATABASE_FILE_PATH))
+	if (Method == FullAssetScan)
 	{
-		Schema::ResetSchemaGeneratorStateAndCleanupFolders();
-	}
+		// Make sure SchemaDatabase is not loaded.
+		if (UPackage* LoadedDatabase = FindPackage(nullptr, *FPaths::Combine(TEXT("/Game/"), *SpatialConstants::SCHEMA_DATABASE_FILE_PATH)))
+		{
+			TArray<UPackage*> ToUnload;
+			ToUnload.Add(LoadedDatabase);
+			UPackageTools::UnloadPackages(ToUnload);
+		}
 
-	TArray<TStrongObjectPtr<UObject>> LoadedAssets;
-	if (bFullScan)
-	{
-		Progress.EnterProgressFrame(80.f);
-		if (!LoadPotentialAssets(LoadedAssets))
+		const USpatialGDKEditorSettings* EditorSettings = GetDefault<USpatialGDKEditorSettings>();
+
+		const FString& PlatformName = EditorSettings->GetCookAndGenerateSchemaTargetPlatform();
+
+		if (PlatformName.IsEmpty())
 		{
-			bSchemaGeneratorRunning = false;
-			LoadedAssets.Empty();
-			CollectGarbage(GARBAGE_COLLECTION_KEEPFLAGS, true);
+			UE_LOG(LogSpatialGDKEditor, Error, TEXT("Empty platform passed to CookAndGenerateSchema"));
 			return false;
 		}
-	}
 
-	// If running from an open editor then compile all dirty blueprints
-	TArray<UBlueprint*> ErroredBlueprints;
-	if (!IsRunningCommandlet())
-	{
-		const bool bPromptForCompilation = false;
-		UEditorEngine::ResolveDirtyBlueprints(bPromptForCompilation, ErroredBlueprints);
-	}
+		FString OptionalParams = EditorSettings->GetCookAndGenerateSchemaAdditionalArgs();
+		OptionalParams += FString::Printf(TEXT(" -targetplatform=%s"), *PlatformName);
 
-	if (bFullScan)
-	{
-		// UNR-1610 - This copy is a workaround to enable schema_compiler usage until FPL is ready. Without this prepare_for_run checks crash local launch and cloud upload.
-		FString GDKSchemaCopyDir = FPaths::Combine(SpatialGDKServicesConstants::SpatialOSDirectory, TEXT("schema/unreal/gdk"));
-		FString CoreSDKSchemaCopyDir = FPaths::Combine(SpatialGDKServicesConstants::SpatialOSDirectory, TEXT("build/dependencies/schema/standard_library"));
-		Schema::CopyWellKnownSchemaFiles(GDKSchemaCopyDir, CoreSDKSchemaCopyDir);
-		Schema::RefreshSchemaFiles(GetDefault<USpatialGDKEditorSettings>()->GetGeneratedSchemaOutputFolder());
-	}
+		FString ProjectPath = FPaths::IsProjectFilePathSet() ? FPaths::ConvertRelativePathToFull(FPaths::GetProjectFilePath()) : FPaths::RootDir() / FApp::GetProjectName() / FApp::GetProjectName() + TEXT(".uproject");
```

Comment:
Surprised there is no helper function for this

Reply:
`FPaths::GetProjectFilePath()`?

Reply:
I just removed the conditional and checked that the project path was set. I do not think we have to bother about the other case.